### PR TITLE
fix(platform): transport `undefined` responses to requestor in request-response communication if `MessageClient#onMessage` and `IntentClient#onIntent` return an Observable

### PR DESCRIPTION
**Pull request by mofogasy**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-microfrontend-platform/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added
- [x] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [x] Fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [ ] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, removal of deprecations, etc)
- [ ] Deps (changes related to updating dependencies)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Issue

Issue Number: #164 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.ts [hunk 1]
```diff
@@ -103,3 +101,11 @@ export class MessageHandler<MSG extends Message, REPLY> {
       }));
   }
 }
+
+function coerceObservable<T>(value: T | Observable<T> | Promise<T>): Observable<T> {
```
Comment (danielwiehl) [inline projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.ts:105]: Wouldn't it be better if we explicitly listed the cases?

```ts
/**
 * Creates an Observable from the result of the callback.
 *
 * If the callback returns no value (`void`), returns `undefined`, or returns a Promise that resolves to `undefined`,
 * the returned observable will complete immediately.
 */
function fromCallbackResult$<T>(value: T | Observable<T> | Promise<T>): Observable<T> {
  if (value === undefined) {
    return EMPTY;
  }
  if (value instanceof Observable) {
    return value;
  }
  if (value instanceof Promise) {
    return from(value).pipe(filter(resolved => resolved !== undefined));
  }
  return of(value);
}
```

Comment (danielwiehl) [reply projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.ts:105]: What do you think of the following commit message?

```
fix(platform): mirror Observable if replying with an Observable in message or intent callback

Previously, the communication was terminated when the Observable emitted `undefined`.
```

File: projects/scion/microfrontend-platform/src/lib/client/messaging/intent-client.ts [hunk 2]
```diff
@@ -139,7 +139,11 @@ export abstract class IntentClient {
    * For each intent received, the specified callback function is called. When used in request-response communication,
    * the callback function can return the response either directly or in the form of a Promise or Observable. Returning a Promise
    * allows the response to be computed asynchronously, and an Observable allows to return one or more responses, e.g., for
-   * streaming data. In either case, when the final response is produced, the handler terminates the communication, completing
+   * streaming data.
+   * If the callback function does not return a value (`void`) or returns `undefined`, no value is transported to the requestor.
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/microfrontend-platform/src/lib/client/messaging/intent-client.ts:143]: done

File: projects/scion/microfrontend-platform/src/lib/client/messaging/message-client.ts [hunk 3]
```diff
@@ -128,7 +128,11 @@ export abstract class MessageClient {
    * For each message received, the specified callback function is called. When used in request-response communication,
    * the callback function can return the response either directly or in the form of a Promise or Observable. Returning a Promise
    * allows the response to be computed asynchronously, and an Observable allows to return one or more responses, e.g., for
-   * streaming data. In either case, when the final response is produced, the handler terminates the communication, completing
+   * streaming data.
+   * If the callback function does not return a value (`void`) or returns `undefined`, no value is transported to the requestor.
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/microfrontend-platform/src/lib/client/messaging/message-client.ts:132]: done

File: projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.spec.ts [hunk 4]
```diff
@@ -268,7 +268,37 @@ describe('Message Handler', () => {
...
+
+      const captor = new ObserveCaptor(bodyExtractFn);
+      Beans.get(MessageClient).request$('topic', 'a').subscribe(captor);
+
+      await captor.waitUntilCompletedOrErrored();
+      await expect(captor.getValues()).toEqual([]);
+      await expect(captor.hasCompleted()).toBeTrue();
+    });
+
+    it('should not ignore `undefined` and `null` values when returning an Observable', async () => {
```
[the remark this answers is not in the document]
Comment (mofogasy) [reply projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.spec.ts:301]: done

File: projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.ts (hunk 1 above)
Comment (mofogasy) [reply projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.ts]: done

**Review by danielwiehl — APPROVED** [review]

**Pull request merged**